### PR TITLE
FUSETOOLS-2768 - logging exception - moved test to an integration tes…

### PR DESCRIPTION
**Pull request by lhein**
…t bundle

Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [x] Did the CI job report a successful build?
- [x] Is the non-happy path working, too?

## Maintainability

- [x] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [x] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

File: syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests.integration/src/main/java/org/fusesource/ide/syndesis/extension/core/tests/integration/SyndesisExtensionIntegrationTestsActivator.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.jboss.tools.foundation.core.plugin.log.IPluginLog;
+import org.jboss.tools.foundation.ui.plugin.BaseUIPlugin;
+
+public class SyndesisExtensionIntegrationTestsActivator extends BaseUIPlugin {
+	
+	private static SyndesisExtensionIntegrationTestsActivator instance = null;
+	
+	public SyndesisExtensionIntegrationTestsActivator() {
+		instance = this;
```
[the remark this answers is not in the document]
Comment (lhein) [reply syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests.integration/src/main/java/org/fusesource/ide/syndesis/extension/core/tests/integration/SyndesisExtensionIntegrationTestsActivator.java]: we ignored that sonar issue usually

**Comment by lhein** [comment]
the point is that this test was formerly in a unit test bundle. There is no way of accessing the plugin log nor the activator in a unit test. thats why I moved it to an integration test.

**Comment by apupier** [comment]
> the point is that this test was formerly in a unit test bundle. There is no way of accessing the plugin log nor the activator in a unit test. thats why I moved it to an integration test.

the log was accessed only if there is an exception, as the file is now available online, there are no more errors and so the plugin log is not used.

**Comment by lhein** [comment]
The log will be accessed whenever there are problems with the access to the file (be it network issues or whatever). We can assume that this will not happen very often but I deem it as NOT impossible.

**Comment by apupier** [comment]
> The log will be accessed whenever there are problems with the access to the file (be it network issues or whatever). We can assume that this will not happen very often but I think it is NOT impossible.

and so you don't want this test to fail if there is a problem? (the file has been deleted from master branch  for instance)

**Comment by lhein** [comment]
why won't the test fail? An error is logged in that case. i am not sure we are talking about the same issue.

**Comment by apupier** [comment]
> why won't the test fail? An error is logged in that case. i am not sure we are talking about the same issue.

the test will not fail if the file has been deleted from master branch for instance because it will use the "fallback" versions.
So there will be a log but the test will still be in success.

By moving the test to Integration instead of Unit, the test is no more testing the same thing. My question are: are you aware of that? Do you think that it is more valuable this way? Should we write another test/improve test for testing the other usecase?

**Pull request closed without merging**